integration/aggregate_stats_compat_test.go

```diff
@@ -23,74 +23,66 @@ import (
 
 	"github.com/FerretDB/FerretDB/integration/setup"
 	"github.com/FerretDB/FerretDB/integration/shareddata"
+	"github.com/FerretDB/FerretDB/internal/types"
 )
 
-func TestAggregateCompatCollStats(tt *testing.T) {
-	tt.Parallel()
+func TestAggregateCompatCollStats(t *testing.T) {
+	t.Parallel()
+
+	s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+		Providers:                []shareddata.Provider{shareddata.ArrayDocuments},
+		AddNonExistentCollection: true,
+	})
+	ctx, targetCollections, compatCollections := s.Ctx, s.TargetCollections, s.CompatCollections
 
 	for name, tc := range map[string]struct {
-		skip       string                   // skip test for all handlers, must have issue number mentioned
-		collStats  bson.D                   // required
+		pipeline bson.A // required
+
 		resultType compatTestCaseResultType // defaults to nonEmptyResult
+		skip       string                   // skip test for all handlers, must have issue number mentioned
 	}{
-		"NilCollStats": {
-			collStats:  nil,
-			resultType: emptyResult,
-		},
```

Comment:
Test cases that produce error are moved to `TestAggregateCollStatsCommandErrors` to check error message